internal/engine/executor.go

```diff
@@ -240,10 +241,21 @@ func (e *Executor) evalEntityEvent(
 			defer e.updateLockLease(ctx, *inf.ExecutionID, evalParams)
 
 			// Evaluate the rule
-			evalParams.SetEvalErr(rte.Eval(ctx, inf, evalParams))
-
-			// Perform actions, if any
-			evalParams.SetActionsErr(ctx, rte.Actions(ctx, inf, evalParams))
+			evalErr := rte.Eval(ctx, inf, evalParams)
+			evalParams.SetEvalErr(evalErr)
+
+			if evalErr == nil {
+				// Perform actions, if any
+				evalParams.SetActionsErr(ctx, rte.Actions(ctx, inf, evalParams))
```

Comment:
Hmm, doesn't this mean that we perform the action if evaluation succeeded and the entity matches the profile? Don't we want to run the actions on `errors.Is(err, ErrEvaluationFailed)` ?

Reply:
duh, thanks, I missed this.